### PR TITLE
Mark z_gettotalbalance and dumpwallet as deprecated

### LINKED ISSUE #5177: Deprecate old address-related RPCs
- ` getnewaddress`
  - UAs will generate P2PKH address components internally, from a key tree that is domain separated from any addresses that might be returned by this RPC.
- `z_getnewaddress`
  - It has the semantics that each call generates a new spend authority for a new logical pool of funds.
  - Its name indicates it fetches a new address rather than a new key, which complicates address diversification.
  - It allows selecting a specific pool, but for UAs we won't be exposing that.
- `z_gettotalbalance`
  - Deprecate the old fields (but keep them for now, including Orchard balances in "private")
  - API itself is fine (we will extend the dictionary output with new fields)

### LINKED ISSUE #5509: Deprecate `dumpwallet` RPC
This RPC method is a loss-of-funds hazard because it only dumps transparent keys, which is quite unintuitive. Users should always use `z_exportwallet` instead.

### PR DESCRIPTION
**Pull request by steven-ecc**
Closes #5177, closes #5509.

### CODE REVIEW

File: src/wallet/rpcdump.cpp [hunk 1]
```diff
@@ -547,6 +547,7 @@ UniValue dumpwallet(const UniValue& params, bool fHelp)
     if (fHelp || params.size() != 1)
         throw runtime_error(
             "dumpwallet \"filename\"\n"
+            "\nDEPRECATED\n"
```
Comment (daira) [inline src/wallet/rpcdump.cpp:550]: Please add a suggestion to use `z_dumpwallet` instead.

File: src/wallet/rpcwallet.cpp [hunk 2]
```diff
@@ -3892,6 +3892,7 @@ UniValue z_gettotalbalance(const UniValue& params, bool fHelp)
     if (fHelp || params.size() > 2)
         throw runtime_error(
             "z_gettotalbalance ( minconf includeWatchonly )\n"
+            "\nDEPRECATED\n"
```
Comment (daira) [inline src/wallet/rpcwallet.cpp:3895]: Please add a suggestion to use `z_getbalanceforaccount` (if I'm remembering the name correctly) instead.

**Review by daira — CHANGES_REQUESTED** [review]
Please add the suggestions for alternative methods.

File: src/wallet/rpcdump.cpp (hunk 1 above)
Comment (str4d) [reply src/wallet/rpcdump.cpp:550]: `z_exportwallet` is the RPC name

**Review by LarryRuane — DISMISSED** [review]
utACK after applying Daira's suggestions

**Review by LarryRuane — DISMISSED** [review]
reACK 512c862a5f2006aee5d89f07b2667cf13cd4ef94

File: src/wallet/rpcwallet.cpp [hunk 3]
```diff
@@ -3892,6 +3892,7 @@ UniValue z_gettotalbalance(const UniValue& params, bool fHelp)
     if (fHelp || params.size() > 2)
         throw runtime_error(
             "z_gettotalbalance ( minconf includeWatchonly )\n"
+            "\nDEPRECATED. Please use the z_getbalanceforaccount RPC instead. \n"
```
Comment (daira) [inline src/wallet/rpcwallet.cpp:3895]:
```suggestion
            "\nDEPRECATED. Please use the z_getbalanceforaccount RPC instead.\n"
```

**Review by daira — APPROVED** [review]
utACK (I committed a cosmetic whitespace change).

**Comment by daira** [comment]
Oh, maybe that change wasn't worth waiting for another CI run. Oh well.

**Pull request merged**